node/src/components/consensus/era_supervisor/era.rs

```diff
@@ -161,6 +163,68 @@ impl<I> Era<I> {
     pub(crate) fn set_paused(&mut self, paused: bool) {
         self.consensus.set_paused(paused);
     }
+
+    /// Computes rewards and returns the era report.
+    pub(crate) fn create_report(
+        &self,
+        terminal_block_data: TerminalBlockData<ClContext>,
+        timestamp: Timestamp,
+        protocol_config: &ProtocolConfig,
+    ) -> EraReport<PublicKey> {
+        let TerminalBlockData {
+            inactive_validators,
+        } = terminal_block_data;
+
+        // The total rewards are the ideal number of blocks (i.e. one per minimum-length round) in
+        // the actual duration of the era, times the BLOCK_REWARD constant, independent of how many
+        // blocks were actually created.
+        let era_duration = timestamp.saturating_diff(self.start_time);
+        let min_round_length = protocol_config.highway_config.min_round_length();
+        let number_of_min_length_rounds = era_duration / min_round_length;
+        let total_rewards: u64 = number_of_min_length_rounds
+            .checked_mul(BLOCK_REWARD)
+            .unwrap_or_else(|| {
+                // With BLOCK_REWARD set to one trillion and a round length of 1 minute, a single
+                // era would have to last 35 years before the total rewards would overflow a u64.
+                error!(%era_duration, %min_round_length, %BLOCK_REWARD, "total rewards overflow");
+                u64::MAX
+            });
+
+        let is_active = |public_key: &PublicKey| {
+            !(total_rewards == 0
```

Comment:
If `total_rewards == 0` then `is_active` returns `false` for all validators and all the rewards computations will be empty/zero. Would it make sense to exit early in that case? Also, the only reason for `total_rewards` to be 0 is if `number_of_min_length_rounds` was 0, I guess that's only possible for the "phantom eras"?

Reply:
Yes, and even for those with the current implementation we wouldn't end up in this code path.

Reply:
Returns early in https://github.com/casper-network/casper-node/pull/1950/commits/8a8090c72da2779d476924de970514641e6f93e7.